mantidproject/mantid
# VSI performance checks

Description of work.

This applies changes suggested by the [clang-tidy performance-* checks](http://clang.llvm.org/extra/clang-tidy/checks/list.html). Reviewing these changes, I found a few more places to reduce the number of unnecessary copies.  

I don't think these changes are in hot loops that would be obvious to end users. 

**To test:**

Review code changes and builds status. If you notice a significant performance improvement, let's add it to the release notes.



There is no GitHub issue associated with this pull request

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
